ansible_base/lib/utils/encryption.py

```diff
@@ -37,34 +38,62 @@ def __init__(self):
         self._encryption_key = self.key[32:]
         self._backend = default_backend()
 
+    def extract_data(self, raw_data: str) -> str:
+        # Sequential data extract for code reuse in is_encrypted_string and decrypt_string
+        raw_data = raw_data[len(ENCRYPTED_STRING) :]
+
+        # If the encrypted string contains a UTF8 marker, discard it
+        if raw_data.startswith('UTF8$'):
+            raw_data = raw_data[len('UTF8$') :]
+
+        # Extract the algorithm and ensure its what we expect
+        algo, data = raw_data.split('$', 1)
+        if algo != ENCRYPTION_METHOD:
+            raise ValueError(f'Unsupported algorithm: {algo}')
+
+        return data
+
+    def is_encrypted_string(self, value: str, invalid_algo_is_fatal: bool) -> bool:
+        # Ensure input is a string already encrypted by our algorithm
+        # by comparing with the decrypted and re-encrypted values
+        if not value.startswith(ENCRYPTED_STRING):
+            return False
+
+        try:
+            data = self.extract_data(value)
+        except ValueError as ve:
+            # value error is raised when algorithm is unsupported
+            if invalid_algo_is_fatal:
+                raise ve
+
+        # Check if data is base64 encoded
+        # **important** Note: this can fail because some strings resemble base 64 encoded ones
+        # **important** for instance, if user's secret is $encrypted$UTF8$AESCBC$junk, the code will break
+        # use regex pattern: ^([A-Za-z0-9+/]{4})* means the string starts with 0 or more base64 groups.
+        # ([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$ means
+        # the string ends in one of three forms: [A-Za-z0-9+/]{4}, [A-Za-z0-9+/]{3}= or [A-Za-z0-9+/]{2}==.
+        reg = r'^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$'
```

Comment:
`junk` [matches your regular expression](https://pythex.org/?regex=(%5BA-Za-z0-9%2B%2F%5D%7B4%7D)*(%5BA-Za-z0-9%2B%2F%5D%7B3%7D%3D%7C%5BA-Za-z0-9%2B%2F%5D%7B2%7D%3D%3D)%3F%24&test_string=junk&ignorecase=0&multiline=0&dotall=0&verbose=0). 

I like where you are going with this though, if we are able to introducing a regex for testing data can we just regex this whole method?
`^\$encrypted\$(UTF8\$)?AESCBC\$<your b64 check>$` (obviously we would want to put variables in there instead of hard coding.